ex04.ex

```diff
@@ -74,10 +82,24 @@ defmodule Ex04 do
   helps you do that. And, if you use that function, what does it return? That
   return value will be the thing you have to manipulate.
   """
-
-  def even_odd . . . "your code"
-
-
+  
```

Comment:
## -4

It works, but there's a bunch of needless duplication, both in the code and in the fact you process the list twice.

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &partition_even_odd/2)
    |> reverse_two_lists
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~